pkg/controller/kubedirectorcluster/members.go

```diff
@@ -280,10 +330,83 @@ func handleReadyMembers(
 				)
 				return
 			}
+
+			memberVersion := m.StateDetail.LastConnectionVersion
+
+			shared.LogInfo(
+				reqLogger,
+				cr,
+				shared.EventReasonCluster,
+				fmt.Sprintf("memberVersion for pod %s is %d", m.Pod, memberVersion),
+			)
+
+			if memberVersion < connectionsVersion {
+				shared.LogInfo(
+					reqLogger,
+					cr,
+					shared.EventReasonCluster,
+					fmt.Sprintf("--reconnect will be called here for pod : %s", m.Pod),
+				)
+
+				//*****
+				// currVersion = cr.Annotations["cluster"]
+
+				// if cr.Annotations[m.Pod] == nil
+				// 		then, initialize new key and value cr.Annotations[m.Pod] = currVersion
+
+				// if cr.Annotations[m.Pod] < currVersion
+				// it means that the Pod wasn't updated after conns changed
+				// call --reconnect and cr.Annotations[m.Pod]++.
+				// We don't do cr.Annotations[m.Pod]=currVersion because if the version changes when this is being updated,
+				// we want to call --reconnect again
+
+				// if cr.Annotations[m.Pod] > currVersion
+				// We dont want such a scenario, error out
+
+				//if cr.Annotations[m.Pod] == currVersion
+				// no action, continue
+
+				//*****
+
+				containerID := m.StateDetail.LastConfiguredContainer
+				cmd := fmt.Sprintf(appPrepConfigRunCmdTest, containerID)
+
+				shared.LogInfo(
+					reqLogger,
+					cr,
+					shared.EventReasonCluster,
+					fmt.Sprintf("Executing command %s", cmd),
+				)
+				cmdErr := executor.RunScript(
+					reqLogger,
+					cr,
+					cr.Namespace,
+					m.Pod,
+					m.StateDetail.LastConfiguredContainer,
+					executor.AppContainerName,
+					"app config",
+					strings.NewReader(cmd),
+				)
+				if cmdErr != nil {
+					shared.LogInfo(
+						reqLogger,
+						cr,
+						shared.EventReasonCluster,
```

Comment:
if we error out, should we put member in error state? we do in expansion case I think but should we put here? what do you think?

Reply:
the member now goes in the same state as it would be when configmeta doesn't get injected, albeit with a different error message